### PR TITLE
Update Azure.ResourceManager.Kusto to 2022-07 version

### PR DESCRIPTION
**Pull request by t-abebchuk**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by Yao725 — CHANGES_REQUESTED** [review]
Hi @t-abebchuk, I have fixed the backward compatibility issue and also made some minor changes in your PR. Now the build should succeed and you can go ahead with tests.

File: sdk/kusto/Azure.ResourceManager.Kusto/tests/KustoManagementTestBase.cs [hunk 1]
```diff
@@ -25,17 +56,119 @@ protected KustoManagementTestBase(bool isAsync)
...
+
+            Cluster = (await clusterCollection.CreateOrUpdateAsync(WaitUntil.Completed, ClusterName, clusterData))
+                .Value;
+
+            return Cluster;
+        }
+
+        public async Task<KustoDatabaseResource> GetDatabase(KustoClusterResource cluster)
+        {
+            if (Database is not null)
```
[the remark this answers is not in the document]
Comment (t-abebchuk) [reply sdk/kusto/Azure.ResourceManager.Kusto/tests/KustoManagementTestBase.cs:106]: Shared for efficiency

File: sdk/kusto/Azure.ResourceManager.Kusto/CHANGELOG.md [hunk 2]
```diff
@@ -1,15 +1,26 @@
 # Release History
 
-## 1.1.0-beta.1 (Unreleased)
+## 1.1.0-beta.1 (2020-10-23)
```
Comment (ArthurMa1978) [inline sdk/kusto/Azure.ResourceManager.Kusto/CHANGELOG.md:3]: Please update the release day for release

Comment (ArthurMa1978) [reply sdk/kusto/Azure.ResourceManager.Kusto/CHANGELOG.md:3]: Also please flip the pr from `draft` to `ready for review`

File: sdk/kusto/test-resources.bicep [hunk 3]
```diff
@@ -0,0 +1,85 @@
+param id string
```
Comment (Yao725) [inline sdk/kusto/test-resources.bicep:1]: As we have deprecated track 1 SDK, I'd like to know why we still need all these changes in `Microsoft.Azure.Management.Kusto` directory.

Comment (sagivf) [reply sdk/kusto/test-resources.bicep:1]: I'm not accentually sure what these are for, but I don't think it's in the Microsoft.Azure.Management.Kusto directory.

Comment (Yao725) [reply sdk/kusto/test-resources.bicep:1]: Oh, sorry, I accidentally misread it

Let me clarify this, for this PR, I think the changes that we need should only happen in the directory `sdk/kusto/Azure.ResourceManager.Kusto`, could we revert other changes?

Comment (t-abebchuk) [reply sdk/kusto/test-resources.bicep:1]: As your test framework mentions https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/TestResources/New-TestResources.ps1.md, it is possible to setup an environment for the tests by using the New-TestResources.ps1 command feeding it a bicep file. The location of the bicep file is expected to be under sdk/libraryName directory. I'm not sure why this is problematic.

File: sdk/kusto/Azure.ResourceManager.Kusto/tests/Azure.ResourceManager.Kusto.Tests.csproj [hunk 4]
```diff
@@ -2,4 +2,14 @@
...
   </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="SessionRecords" />
+    <Folder Include="SessionRecords\" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.ResourceManager.Network" />
+    <PackageReference Include="System.Linq.Async" />
```
[the remark this answers is not in the document]
Comment (t-abebchuk) [reply sdk/kusto/Azure.ResourceManager.Kusto/tests/Azure.ResourceManager.Kusto.Tests.csproj:13]: First package was not used and is removed.
The second package Linq is used such as in calling FirstOrDefaultAsync() on IAsyncEnumerable objects is KustoTestBase.

File: sdk/kusto/test-resources.bicep (hunk 3 above)
Comment (t-abebchuk) [reply sdk/kusto/test-resources.bicep:1]: This file is for our personal use. It is run once before recording. The point of running this file is to create a setup script that creates all the resources we need to create a test environment without having to write a bunch of code to create it.

File: sdk/kusto/setup.ps1 [hunk 5]
```diff
@@ -0,0 +1,30 @@
+Import-Module Az.Kusto
```
Comment (Yao725) [inline sdk/kusto/setup.ps1:1]: I believe this is only for the tests in `Azure.ResourceManager.Kusto`. If that's correct, could you please do me a favor to make the project and file structure clearer? I think what we can do here include:

- Create a new folder `Prerequisite` under `sdk/kusto/Azure.ResourceManager.Kusto/tests`.
- Move the `setup.ps1`, `test-resources.bicep` and the folder `data` to `Prerequisite`
- Create a new markdown file under `Prerequisite` to tell the coworkers what setup.ps1 is used for, what they need to install and update when they want to run the script. Here is one [example](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/tests/prerequisites/prerequisite%20readme.md) from another RP.

File: sdk/kusto/test-resources.bicep (hunk 3 above)
Comment (Yao725) [reply sdk/kusto/test-resources.bicep:1]: Ok, that makes sense to me, thanks for you explanation.

**Review by Yao725 — APPROVED** [review]
LGTM

**Pull request merged**